formula/src/main/java/com/instacart/formula/internal/FormulaManagerImpl.kt

```diff
@@ -36,22 +36,23 @@ internal class FormulaManagerImpl<Input, State : Any, Output>(
     private var state: State = formula.initialState(initialInput)
     private var pendingRemoval: MutableList<FormulaManager<*, *>>? = null
 
-    private fun handleTransition(transition: Transition<State>, wasChildInvalidated: Boolean) {
+    private var childTransitionListener: TransitionListener? = null
+
+    private fun handleTransition(transition: Transition<State>) {
         if (terminated) {
             // State transitions are ignored, only side effects are passed up to be executed.
-            transitionListener.onTransition(transition.effects, true)
+            transitionListener.onTransition(transition, true)
             return
         }
 
-        this.state = transition.state ?: this.state
+        this.state = when (transition) {
```

Comment:
Not sure how much we prefer optimizations vs readability, but you can always do the following to remove a few extra lines of bytecode to load/get the `this.state` field.
```
        if (transition is Transition.Stateful) {
            this.state = transition.state
        }
```

Reply:
This looks better, updating...